lib/xapi-stdext-date/date.mli

```diff
@@ -18,30 +18,50 @@
 (** An ISO-8601 date/time type. *)
 type iso8601
```

Comment:
Conceptually, ISO8601 can represent a time not just in UTC but other time zones as well. I believe we  only generate UTC-based ISO8601 timestamps and only convert from such time stamps. The type could be called iso8601utc to reflect that.

Reply:
In tests we need to be careful that they could only work because our local time is UTC but would fail in a different time zone.

Reply:
It's a weird name for an abstract type anyway, because in my mind ISO-8601 is just a string representation, i.e. a `string` coming out of a `to_string` function. How about having a `type t` here, which in the ml file is `type t = Ptime.t`?

Reply:
I'd like to use the type t as an alias for the time being (code generation in xapi uses this type, so I'd rather have both types available to ease transition.

about the definition of type t I can try, but it's encoded in another way to be able to print the Z at the end when the local time is in UTC, hence
```ocaml
type iso8601 = Ptime.date * Ptime.time * print_timezone
```

Reply:
Oh I read to quickly. I thought there was `type iso8601 = string`, but there is `type rfc822 = string` instead (and `type iso8601` as you say).

Reply:
I've unified both types to a single `t` with appropriate conversion functions :)

Reply:
Perfect!